packages/core/workflows-sdk/src/utils/_playground.ts

```diff
@@ -53,6 +55,28 @@ workflow2()
     console.log(res.result)
   })
 
+type MyWorkflowHooks = {
+  beforeCreate: (id: string) => string
+}
+
+const workflowWithHooks = createWorkflowWithHooks(
```

Comment:
hint: We are able here to maintain the chain of type coming from the inner workflow, which means the inputs are still typed etc